bitcoin/src/bip32.rs

```diff
@@ -125,6 +125,39 @@ pub enum ChildNumber {
         index: u32,
     },
 }
+
+/// NormalChildNumber is a ChildNumber that is not hardened
+/// made only to be used internally in `derive_pub()` and `NormalChildIterator`
+#[derive(Copy, Clone, PartialEq, Eq, Debug, PartialOrd, Ord, Hash)]
+pub struct NormalChildNumber(pub u32);
+
+/// A iterator over the children of a DerivationPath trying to convert them into NormalChildNumber
+/// made only to be used internally in `derive_pub()`
+pub struct NormalChildIterator {
+    base: Vec<NormalChildNumber>,
+}
+impl Into<NormalChildIterator> for DerivationPath {
+    fn into(self) -> NormalChildIterator {
+        NormalChildIterator {
+            base: self
+                .0
+                .into_iter()
+                .filter_map(|child| {
+                    if let ChildNumber::Normal { index } = child {
+                        Some(NormalChildNumber(index))
+                    } else {
+                        None
```

Comment:
Silently changing the path is absolutely unacceptable since that will lead into unspendable coins. We have to error.

Reply:
Do you want a log! ? Or just returning zero ? We cannot return a error because its infalible.

Reply:
We could return an error, because it is infallible in the context of BIP32, see https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki#public-parent-key--public-child-key

> The function CKDpub((K<sub>par</sub>, c<sub>par</sub>), i) &rarr; (K<sub>i</sub>, c<sub>i</sub>) computes a child extended public key from the parent extended public key. > It is only defined for non-hardened child keys.
> * Check whether i ≥ 2<sup>31</sup> (whether the child is a hardened key).
> ** If so (hardened child): return failure

See the failure above

Reply:
Oh ok. Making a change that results path.try_into() is indeed more safe.
But just moving errors im the end.

> My reply was mistaken, fixed.